File: ReviewsJuegos/Form1.vb
﻿Imports System.Collections.ObjectModel
Imports System.Data.Common
Imports System.Data.SqlClient

Public Class Form1
    Private query As String
    Private auto As Boolean
    ' Create a constructor
    Public Sub New()
        auto = True
        InitializeComponent()
        DataGridView1.AutoSizeColumnsMode = DataGridViewAutoSizeColumnsMode.DisplayedCells


        query = "SELECT Id,NombreJuego FROM Juego"
        ComboBoxJuego.DataSource = ConexionSQL.SelectQuery(query)
        ComboBoxJuego.DisplayMember = "NombreJuego"
        ComboBoxJuego.ValueMember = "id"
        auto = False
        DataGridView1.ReadOnly = True

        query = "SELECT * FROM VW_JuegoReview"
        DataGridView1.DataSource = ConexionSQL.SelectQuery(query)



    End Sub





    Private Sub ComboBoxJuego_SelectedIndexChanged(sender As Object, e As EventArgs) Handles ComboBoxJuego.SelectedIndexChanged
        If ComboBoxJuego.Text Is Nothing Then
            Return
        End If
        query = "SELECT * FROM VW_JuegoReview
where nombrejuego='" & ComboBoxJuego.Text & "'"
        DataGridView1.DataSource = ConexionSQL.SelectQuery(query)
    End Sub

    Private Sub BtnBuscar_Click(sender As Object, e As EventArgs) Handles BtnBuscar.Click
        Dim buscarTexto As String = TxtJuego.Text.Trim()
        Dim query = ""
        query = "SELECT * FROM VW_JuegoReview
WHERE nombrejuego LIKE '%" & buscarTexto & "%'"
        ' Ejecutar la consulta SQL y mostrar los resultados en el DataGridView
        Dim dt = ConexionSQL.SelectQuery(query)

        DataGridView1.DataSource = dt



    End Sub

    Private Sub TxtId_TextChanged(sender As Object, e As EventArgs) Handles TxtId.TextChanged
        Dim buscarId As Integer

        ' Verificar si el texto en el TxtId es un número válido
        If Integer.TryParse(TxtId.Text.Trim(), buscarId) Then
            ' Iterar sobre las filas del DataGridView para encontrar la que corresponde al ID buscado
            For Each row As DataGridViewRow In DataGridView1.Rows
                If Convert.ToInt32(row.Cells("IdJuego").Value) = buscarId Then
                    ' Seleccionar la fila encontrada y salir del bucle
                    DataGridView1.CurrentCell = row.Cells(0)
                    Exit For
                End If
            Next
        Else
            ' Si el texto no es un número válido, limpiar la selección del DataGridView
            DataGridView1.ClearSelection()
        End If
    End Sub


    Private Sub btnEliminarJuego_Click(sender As Object, e As EventArgs) Handles BtnEliminar.Click
        Dim connectionString As String = "Data Source=localhost;Initial Catalog=ReviewJuego;Integrated Security=True"
        Dim connection As New SqlConnection(connectionString)

        Try
            connection.Open()
            Dim command As New SqlCommand("SP_EliminarJuego", connection)
            command.CommandType = CommandType.StoredProcedure

            ' Suponiendo que el procedimiento almacenado toma el ID del juego como parámetro
            command.Parameters.AddWithValue("@IdJuego", TxtId.Text)

            command.ExecuteNonQuery()
            MessageBox.Show("Juego eliminado correctamente.")
        Catch ex As Exception
            MessageBox.Show("Error al eliminar el juego: " & ex.Message)
        Finally
            connection.Close()
        End Try
    End Sub


    Private Sub ActualizarDataGridView()
        ' Actualizar el DataGridView con la lista actualizada de juegos
        query = "SELECT * FROM VW_JuegoReview"
        DataGridView1.DataSource = ConexionSQL.SelectQuery(query)
    End Sub


    Private Sub BtnAgregar_Click(sender As Object, e As EventArgs) Handles BtnAgregar.Click
        FormularioAgregar.Show()
    End Sub

    Private Sub BtnActualizar_Click(sender As Object, e As EventArgs) Handles BtnActualizar.Click
        FormularioActualizar.Show()
    End Sub

    Private Sub BtnRefrescar_Click(sender As Object, e As EventArgs) Handles BtnRefrescar.Click
        Me.Refresh()
    End Sub
End Class

File: ReviewsJuegos/Form1.cs
using System;
using System.Data;
using System.Data.SqlClient;
using System.Windows.Forms;

namespace ReviewsJuegos
{

    public partial class Form1
    {
        private string query;
        private bool auto;
        // Create a constructor
        public Form1()
        {
            auto = true;
            InitializeComponent();
            DataGridView1.AutoSizeColumnsMode = DataGridViewAutoSizeColumnsMode.DisplayedCells;


            query = "SELECT id,Genero FROM Genero";
            ComboBoxJuego.DataSource = ConexionSQL.SelectQuery(query);
            ComboBoxJuego.DisplayMember = "Genero";
            ComboBoxJuego.ValueMember = "id";
            auto = false;
            DataGridView1.ReadOnly = true;

            query = "SELECT * FROM VW_JuegoReview";
            DataGridView1.DataSource = ConexionSQL.SelectQuery(query);



        }





        private void ComboBoxJuego_SelectedIndexChanged(object sender, EventArgs e)
        {
            if (ComboBoxJuego.Text is null)
            {
                return;
            }
            query = @"SELECT * FROM VW_JuegoReview
where Genero='" + ComboBoxJuego.Text + "'";
            DataGridView1.DataSource = ConexionSQL.SelectQuery(query);
        }

        private void BtnBuscar_Click(object sender, EventArgs e)
        {
            string buscarTexto = TxtJuego.Text.Trim();
            string query = "";
            query = @"SELECT * FROM VW_JuegoReview
WHERE nombrejuego LIKE '%" + buscarTexto + "%'";
            // Ejecutar la consulta SQL y mostrar los resultados en el DataGridView
            var dt = ConexionSQL.SelectQuery(query);

            DataGridView1.DataSource = dt;



        }

        private void TxtId_TextChanged(object sender, EventArgs e)
        {
            int buscarId;

            // Verificar si el texto en el TxtId es un número válido
            if (int.TryParse(TxtId.Text.Trim(), out buscarId))
            {
                // Iterar sobre las filas del DataGridView para encontrar la que corresponde al ID buscado
                foreach (DataGridViewRow row in DataGridView1.Rows)
                {
                    if (Convert.ToInt32(row.Cells["IdJuego"].Value) == buscarId)
                    {
                        // Seleccionar la fila encontrada y salir del bucle
                        DataGridView1.CurrentCell = row.Cells[0];
                        break;
                    }
                }
            }
            else
            {
                // Si el texto no es un número válido, limpiar la selección del DataGridView
                DataGridView1.ClearSelection();
            }
        }


        private void btnEliminarJuego_Click(object sender, EventArgs e)
        {
            string connectionString = "Data Source=localhost;Initial Catalog=ReviewJuego;Integrated Security=True";
            var connection = new SqlConnection(connectionString);

            try
            {
                connection.Open();
                var command = new SqlCommand("SP_EliminarJuego", connection);
                command.CommandType = CommandType.StoredProcedure;

                // Suponiendo que el procedimiento almacenado toma el ID del juego como parámetro
                command.Parameters.AddWithValue("@IdJuego", TxtId.Text);

                command.ExecuteNonQuery();
                MessageBox.Show("Juego eliminado correctamente.");
            }
            catch (Exception ex)
            {
                MessageBox.Show("Error al eliminar el juego: " + ex.Message);
            }
            finally
            {
                connection.Close();
            }
        }


        private void ActualizarDataGridView()
        {
            // Actualizar el DataGridView con la lista actualizada de juegos
            query = "SELECT * FROM VW_JuegoReview";
            DataGridView1.DataSource = ConexionSQL.SelectQuery(query);
        }


        private void BtnAgregar_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.FormularioAgregar.Show();
        }

        private void BtnActualizar_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.FormularioActualizar.Show();
        }

        private void BtnRefrescar_Click(object sender, EventArgs e)
        {
            query = "SELECT id,Genero FROM Genero";
            ComboBoxJuego.DataSource = ConexionSQL.SelectQuery(query);
            ComboBoxJuego.DisplayMember = "Genero";
            ComboBoxJuego.ValueMember = "id";
        }

        private void Form1_Load(object sender, EventArgs e)
        {

        }

        private void Button1_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.AgregarGenero.Show();
        }

        private void BtnAgregarPlata_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.AgregarPlataforma.Show();
        }

        private void BtnAgDesarrolladora_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.AgregarDesarrolladora.Show();
        }

        private void Button2_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.EliminarDesarrolladora.Show();
        }

        private void Button3_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.EliminarPlataforma.Show();
        }

        private void Button4_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.EliminarGenero.Show();
        }

        private void Button7_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.EditarDesarrolladora.Show();
        }

        private void Button5_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.EditarPlataforma.Show();
        }

        private void Button6_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.EditarGenero.Show();
        }
    }
}